lib/CBattleCallback.cpp

```diff
@@ -1020,8 +1020,28 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
 		multBonus *= 0.5;
 	}
 
-	// TODO attack on petrified unit 50%
 	// blinded unit retaliates
+	auto blinded = info.attackerBonuses->getBonus(Selector::source(Bonus::SPELL_EFFECT, SpellID::BLIND));
+	if(blinded)
+	{
+		switch(blinded->val)
+		{
+			case SecSkillLevel::BASIC:

```

Comment:
Spell effect bonus "val" can`be SecSkillLevel. It is explicitly defined number (defaults to "power" spell attribute).